src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -220,33 +221,73 @@ public function loadClass($class)
 
                 throw new \RuntimeException(sprintf('The autoloader expected class "%s" to be defined in file "%s". The file was found but the class was not in it, the class name or namespace probably has a typo.', $class, $file));
             }
-            if (self::$caseCheck && preg_match('#([/\\\\][a-zA-Z_\x7F-\xFF][a-zA-Z0-9_\x7F-\xFF]*)+\.(php|hh)$#D', $file, $tail)) {
+            if (self::$caseCheck && preg_match('#(?:[/\\\\][a-zA-Z_\x7F-\xFF][a-zA-Z0-9_\x7F-\xFF]*)+\.(?:php|hh)$#D', $file, $tail)) {
```

Comment:
Possessive quantifiers: done. As I'm not used to them, please review carefully (you read correctly, there is no dot in this pattern)